src/dftbp/timedep/linrespcommon.F90

```diff
@@ -2862,37 +2863,27 @@ subroutine calcMatrixSqrt(matIn, spaceDim, memDim, workArray, workDim, matOut, m
     !> Inverse of matrix square root
     real(dp), intent(out) :: matInvOut(:,:)
 
-    !> Workspace array
-    real(dp), intent(out) :: workArray(:)
-
-    !> Size of work array
-    integer :: workDim
-
     real(dp) :: dummyEV(spaceDim)
     real(dp) :: dummyM(spaceDim, spaceDim), dummyM2(spaceDim, spaceDim)
-    integer :: info
     integer :: ii
-    external dsyev, dgemm
 
     dummyM(:,:) = matIn(1:spaceDim, 1:spaceDim)
 
-    call dsyev('V', 'U', spaceDim, dummyM, spaceDim, dummyEV, workArray, workDim, info)
+    call heev(dummyM, dummyEV, 'U', 'V') 
 
     ! Calc. sqrt
     do ii = 1, spaceDim
       dummyM2(:,ii) = sqrt(dummyEV(ii)) * dummyM(:,ii)
     end do
 
-    call dgemm('N', 'T', spaceDim, spaceDim, spaceDim, 1.0_dp, dummyM2, spaceDim, dummyM,&
-        & spaceDim, 0.0_dp, matOut, memDim)
+    call gemm(matOut(:spaceDim,:spaceDim), dummyM2, dummyM, transB='T')
```

Comment:
Probably better to extend the gemm interface to include setting LDA for the matrix, as the change as is creates a temporary copy.

Reply:
You are right. I completely redesigned the weird memory setup in Stratmann. Now temp arrays are avoided and arrays are allocated to their required size. This simplifies a lot of gemm calls. Code is 100 % faster in the limited cases I studied and does not need more memory.